packages/fuel-indexer-macros/src/indexer.rs

```diff
@@ -509,6 +509,12 @@ fn process_fn_items(
                                 let data = bincode::serialize(&abi::TransferOut{ contract_id: id, to, asset_id, amount, pc, is }).expect("Bad encoding,");
                                 decoder.decode_type(ty_id, data);
                             }
+                            Receipt::Revert { id, ra, .. } => {
+                                #contract
+                                let ty_id = abi::Revert::type_id();
+                                let data = bincode::serialize(&abi::Revert{ contract_id: id, ra }).expect("Bad encoding,");
+                                decoder.decode_type(ty_id, data);
+                            }
```

Comment:
Just a thought, but what if we moved the bitmasking from the handler to the codegen itself? For example (using the other suggestion of an `error_val` field in `abi::Revert`):
```
Receipt::Revert { id, ra, .. } => {
    #contract
    let ty_id = abi::Revert::type_id();
    let data = bincode::serialize(&abi::Revert{ contract_id: id, error_val: u64::from(ra & 0xF) }).expect("Bad encoding,");
    decoder.decode_type(ty_id, data);
}
```

It seems improbable that a user would use the very large decimal values directly from the receipt other than to apply a bitmask upon them so if we we do it for them here in the codegen, they wouldn't have to worry about it. We could then add a mapping of the much smaller values and their meanings to a `Revert` page of the Receipts section of the book. Additionally, they could also store the value straight into the database as well.

Furthermore, you wouldn't have to define a `FuelError` enum or implement the `unmask()` operation.

Reply:
Yes, you are right we shouldn't expose this to the user 👌.
I'll update the receipt section of the docs in this PR also.